src/core/src/queue.rs

```diff
@@ -58,19 +58,51 @@ impl<Q: CommandQueue> DerefMut for GeneralQueue<Q> {
     }
 }
 
-impl<Q: CommandQueue> Into<GraphicsQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> GraphicsQueue<Q> {
-        GraphicsQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for GraphicsQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> GraphicsQueue<Q> {
+        GraphicsQueue(queue.0)
     }
 }
-impl<Q: CommandQueue> Into<ComputeQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> ComputeQueue<Q> {
-        ComputeQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for ComputeQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> ComputeQueue<Q> {
+        ComputeQueue(queue.0)
     }
 }
-impl<Q: CommandQueue> Into<TransferQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> TransferQueue<Q> {
-        TransferQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for TransferQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> TransferQueue<Q> {
+        TransferQueue(queue.0)
+    }
+}
+
+impl<'a, Q: CommandQueue> From<&'a GeneralQueue<Q>> for &'a GraphicsQueue<Q> {
```

Comment:
These conversion implementations are mostly for convenience. I noticed during writing the App part that we probably can't relay on APIs to expose at least one GeneralQueue, but at least a GeneralQueue or a GraphicsQueue (if the backend doesnt support compute). Due to typesafety it becomes quite verbose to select the correct queue. With these impls queue types can be `downcasted` which makes it easier.